### DISCUSSION TITLE
5.3 Release Planning

### DISCUSSION THREAD

**Discussion by mattijn** [vega/altair#3333] [General]
Kicking in a issue in which we can discuss what we would like to include in a 5.3 release. 

This release also can be an opportunity for someone else getting the experience to cut a release. Hopefully this experience contribute to smoothing out the process for future releases.

**Comment by jonmmease** [comment]
I think I'd like to get https://github.com/altair-viz/altair/issues/2818 in. @ChristopherDavisUCI and I submitted a tutorial proposal for SciPy, and it would be great to have a simplified installation instructions.

**Comment by binste** [comment]
Happy to cut the release this time! @mattijn Sorry if we had this discussion before but I can't find any notes on it except for https://github.com/altair-viz/altair/issues/2793. Do you have anything specific in mind when you talk about "smoothing out the process"? Is https://github.com/altair-viz/altair/blob/main/RELEASING.md the most up-to-date documentation of the process?

+1 on getting #2818 in. I might try to get some small doc improvements in as well but we don't have to wait for those.

**Reply by mattijn** [reply]
Nice that you like to do the release! Yes, for me, that issue you are referring to is the end-goal, but I don't think we can make it in once. Probably we have to do some changes to the release process in order to have it a bit more fluent/smooth in order to automate it by a GitHub action.

**Comment by mattijn** [comment]
Is https://github.com/altair-viz/altair/issues/3318 something realistic to be included in Altair @jonmmease? If yes, shall we try to include it in a 5.3 release or is that too optimistic?

I had another question on types in the autocomplete modals @binste. In Altair everything is more or less imported in the root of Altair.
```
import altair as alt
alt.<anything>
```
But oftenly I see types being referred to their deeply nested origin within Altair codebase. Can we try to improve this? I can open an issue for this with a screenshot.

**Reply by binste** [reply]
I can't quite follow. Do you mean if we can reduce how many objects are importable from top-level altair? maybe this is related to your question: #2918

**Reply by mattijn** [reply]
I've created the following issue explaining what I meant, https://github.com/altair-viz/altair/issues/3335

**Comment by jonmmease** [comment]
Sorry for the delay, here's the PR with the all dependency group: https://github.com/altair-viz/altair/pull/3354

**Comment by jonmmease** [comment]
We should also update to Vega-Lite 5.17, which means I need to do a vl-convert release. https://github.com/vega/vega-lite/releases/tag/v5.17.0

**Reply by binste** [reply]
Agree! I can do the update of Altair on the weekend if no one gets to it before that.

**Reply by joelostblom** [reply]
If we can get https://github.com/altair-viz/altair/pull/3358 in before we update that would be convenient. It should hopefully be mergeable in the next couple of days.

**Reply by jonmmease** [reply]
Just published vl-convert 1.3.0 with Vega-Lite 5.17.0 support, and the latest Vega 5.28.0. conda-forge packages are still building, but everything is up on PyPI.

**Reply by binste** [reply]
Upgrade to VL 5.17 is ready to be reviewed: https://github.com/altair-viz/altair/pull/3367

**Comment by jonmmease** [comment]
I just opened a WIP PR to support displaying charts in external browser tabs (https://github.com/altair-viz/altair/pull/3379). If we like this approach, it may be nice to get this in for 5.3. But if you were ready to do the release right away @binste, no need to wait on it.

**Reply by binste** [reply]
I agree, that would be great to get in as well! I was thinking about doing the release next weekend as I'd have time but could also do it the weekend after. Let's see how it looks next weekend :)

**Reply by jonmmease** [reply]
Next weekend sounds like a good goal. I'll try to wrap up this PR by mid week (and I'll try to not start anything new until after the release 😄)

**Reply by binste** [reply]
Hi @jonmmease, just wanted to check in if you'd want #3379 included in 5.3.0. I'd have time to make the release tomorrow (Saturday) evening (UTC time) or then on Monday or also the following weekend. No rush, just want to make sure we can get the PR in if we think it's ready or we skip it if it needs more time.

**Reply by jonmmease** [reply]
Yeah, I'd like to. It's done from my side. I'll merge it this evening (in 5 hours or so) unless there are other comments.

Edit: I saw this notification before seeing that you and joel already reviewed that PR. Thanks! I'll make the editorial updates and merge this evening.

**Comment by joelostblom** [comment]
Would you all be on board with archiving the altair-viewer and altair-saver repos after we make the 5.3 release? I believe all their functionality is now available in Altair + vl-convert, unless I'm forgetting something? We can add a note in the readme with links to the relevant parts of the docs

**Reply by jonmmease** [reply]
Yes, I think that's a good idea. It will help minimize confusion if the README's for those have a header section that directs to the relevant doc sections for the new way to do things.

**Reply by binste** [reply]
Archiving and README with links sound good to me!

**Reply by joelostblom** [reply]
Great, I have commented on and closed all open issues and PRs in both repos (so you might have ~50 notifications from me...). I will leave them open for 1-2 days before archiving in case someone chimes in with something we have missed.

What about https://github.com/altair-viz/altair_data_server? Is there anything in that repo that is not already covered by the vegafusion widget? Or should we archive that as well?

**Reply by binste** [reply]
Awesome, thank you Joel! To me it seems that the use cases of altair_data_server are covered by vegafusion and the integration with JupyterChart.

**Comment by binste** [comment]
5.3.0 is out! 🥳 Thanks to everyone involved

**Reply by joelostblom** [reply]
Thank you, the release notes look great! So much work has gone into this release and it's cool to see all the PRs and individual contributors (11 in total of which 5 are first time contributors! :tada:)

**Comment by jonmmease** [comment]
Thanks @binste!  I think it would be great to tweet an announcement this week from the vega_vis account. There are a lot of exciting this in this release!  I don't have time today to work on this, but can do some later in the week. Let me know if you have preferences for what topics are highlighted.

**Reply by binste** [reply]
Thanks Jon, that would be cool! The better integration with VegaFusion is definitely worth highlighting :) and whatever else you feel like

**Reply by jonmmease** [reply]
I've started drafting some posts based on the release notes, and tested that I can log into the vega_vis twitter account. I'm planning to make a post tomorrow around 14:00 UTC

**Reply by jonmmease** [reply]
After making the announcement thread at https://twitter.com/vega_vis/status/1775881449968681461, it was suggested that we include this info (release highlights, videos, and images) somewhere else where it's easier to share on platforms other than twitter.

Should I add a "Highlights" section to the GitHub release (https://github.com/altair-viz/altair/releases/tag/v5.3.0) where I'd essentially translate these tweets + videos into sections?  Or are there other ideas for where this should live?

**Reply by joelostblom** [reply]
It would be great to have some screenshots and videos! I think we have done it in one or two of the previous releases, so whoever has time to create some screenshots/animations for each release should feel free to add them to the notes (but not a requirement for the person making the release, although it's good if it happens at the same time I guess in terms of people getting notifications).

I'm don't feel strongly regarding creating a new "Highlight" section or just adding them under the enhancement section as @mattijn  did in 5.1 (and Jake in 4.0). Although I'm leaning bit towards the latter and just adding the most important feature on top with screenshots, but still under the same section called "enhancements" so that we don't have that info twice?

I made the PR section collapsible so that the notes don't become too long .

**Reply by jonmmease** [reply]
Ok, I added the videos to the release notes page and reordered it so that the highlights with videos come first.  Thanks!

**Reply by binste** [reply]
Great work on spreading the word about Altair 5.3, thank you both! I really like the updated release notes and the Twitter thread :)
